src/cpp-common/include/cpp-common/HashMap.h

```diff
@@ -0,0 +1,34 @@
+#ifndef PROPOLIS_HASHMAP_H
+#define PROPOLIS_HASHMAP_H
+
+#include "IHashMap.h"
+#include <type_traits>
+
+template <typename Key, typename MappedType, uint16_t maxSize>
+class HashMap : public IHashMap<Key, MappedType, maxSize> {
+  public:
+    HashMap();
+    ~HashMap() override;
+
+    bool insert(const std::pair<Key, MappedType>& item) override;
+    bool get(Key key, MappedType& item) const override;
+    std::optional<MappedType> at(Key key) const override;
+    bool remove(Key key) override;
+    void clear() override;
+    bool isFull() const override;
+    bool isEmpty() const override;
+    uint16_t getMaxSize() const override;
+
+  private:
+    // Using aligned storage for placement new usage when inserting
+    typename std::aligned_storage<sizeof(std::pair<Key, MappedType>),
```

Comment:
since you already have another array, you could just have another one for the hash instead of a pair, or go all in with the tuples and have a tuple of three values.